packages/web-components/src/components/masthead/masthead-composite.ts

```diff
@@ -1326,6 +1327,49 @@ class DDSMastheadComposite extends HostListenerMixin(LitElement) {
       : unauthenticatedCtaButtons;
   }
 
+  /**
+   * A reference to the dds-masthead element.
+   */
+  @query(`${ddsPrefix}-masthead`)
+  mastheadRef;
+
+  /**
+   * Resize observer to trigger container height recalculations.
+   *
+   * @private
+   */
+  private _heightResizeObserver = new ResizeObserver(
+    this._resizeObserverCallback.bind(this)
+  );
+
+  /**
+   * Prevents resize observer from blocking main thread.
+   */
+  private _resizeObserverThrottle?: NodeJS.Timeout;
+
+  /**
+   * Throttled callback for _heightResizeObserver.
+   */
+  protected _resizeObserverCallback() {
+    if (!this._resizeObserverThrottle) {
+      this._setContainerHeight();
+      this._resizeObserverThrottle = setTimeout(() => {
+        this._resizeObserverThrottle = undefined;
+      }, 100);
+    }
```

Comment:
I think the logic here might be backwards: "if there isn't a timer, set the height and prevent additional changes for 100ms". This would mean you could have up to 100ms of additional resizes without the setContainerHeight method firing.

we probably want something closer to this:
```ts
clearTimeout(this._resizeObserverThrottle);
setTimeout(this._setContainerHeight.bind(this), 100)
```

Then, in a series of rapidly firing events, we wait for an idle period of 100ms and then adjust.